src/main/io/beeper.c

```diff
@@ -360,21 +355,61 @@ static void beeperGpsStatus(void)
     if (!(beeperConfig()->beeper_off_flags & BEEPER_GET_FLAG(BEEPER_GPS_STATUS))) {
         // if GPS 3D fix and at least the minimum number available, then beep out number of satellites
         if (STATE(GPS_FIX) && gpsSol.numSat > GPS_MIN_SAT_COUNT) {
-            uint8_t i = 0;
-            do {
-                beep_multiBeeps[i++] = 5;
-                beep_multiBeeps[i++] = 10;
-            } while (i < MAX_MULTI_BEEPS && gpsSol.numSat > i / 2);
-
-            beep_multiBeeps[i - 1] = 50; // extend last pause
+            unsigned i = 0;
+            i = beep_multiBeepsAdd(i, gpsSol.numSat,
+                                   5,
+                                   10, 50);
             beep_multiBeeps[i] = BEEPER_COMMAND_STOP;
 
-            beeper(BEEPER_MULTI_BEEPS);    //initiate sequence
+            beeper(BEEPER_MULTI_BEEPS);
         }
     }
 }
 #endif
 
+/*
+ * Calculates array position when next to change beeper state is due.
+ */
+enum beeperState_e {
+    BeepOnFirst,
+    BeepOn,
+    BeepOff,
+    BeepDone,
+    BeepError,
+};
+
+static enum beeperState_e beeperSequenceAdvance(timeUs_t currentTimeUs)
+{
+    unsigned origPos = beeperPos;
+    bool wasRepeat = false;
+    while (true) {  // evaluate commands until return statement
+        switch (currentBeeperEntry->sequence[beeperPos]) {
+        case BEEPER_COMMAND_REPEAT:
+            if (wasRepeat) {     // prevent infinite loop on badly formed beep sequence
+                return BeepError;
+            }
+            wasRepeat = true;
+            beeperPos = 0;
+            continue;
+        case BEEPER_COMMAND_STOP:
+            return BeepDone;
+        case 0:
```

Comment:
Perhaps add `#define BEEPER_COMMAND_NULL   0`

Reply:
Maybe, but it is also time - spend 0 * 10ms in that phase ...